### PR TITLE
Add documentation for the NetCDF package module

### PR DESCRIPTION
**Pull request by daviditen**
Add some header documentation for the NetCDF module and C_NetCDF submodule.
Point at the official netCDF web site to document the extern interface.  Add
NetCDF to the Makefile to start generating the documentation for it.

### CODE REVIEW

File: modules/packages/NetCDF.chpl [hunk 1]
```diff
@@ -17,7 +17,29 @@
  * limitations under the License.
  */
 
+/* NetCDF bindings for Chapel
+
+This module implements the C-API for netCDF version 4.6.1.  It is expected
```
Comment (ben-albrecht) [inline modules/packages/NetCDF.chpl]: I tend to prefer documenting things for what they are, and leaving planned features / minor bugs in issues, so I'd lean towards leaving this out. That said, I don't think there is an established practice on this, so I'll leave it up your judgement.

File: modules/packages/NetCDF.chpl [hunk 2]
```diff
@@ -17,7 +17,29 @@
...
+that higher-level functionality will be built on top of this API similar to
+the functions in the :mod:`HDF5` module.
+
+Compiling with netCDF
+---------------------
+
+In order to compile a Chapel program that uses this module, the netCDF library
+must be installed on the system.  The paths to the ``netcdf.h`` header file
+and netCDF library must be known to the compiler, either by finding them in
+a default search path, or by using the ``-I`` and ``-L`` compiler arguments.
```
Comment (ben-albrecht) [inline modules/packages/NetCDF.chpl]: Bonus points for a code block showing a compilation with the correct `-I`, `-L`, and `-l` flags, similar to other modules with third-party dependencies.

File: modules/Makefile [hunk 3]
```diff
@@ -106,6 +106,7 @@ PACKAGES_TO_DOCUMENT = \
 	packages/LAPACK.chpl \
 	packages/LinearAlgebra.chpl \
 	packages/MPI.chpl \
+        packages/NetCDF.chpl \
```
Comment (ben-albrecht) [inline modules/Makefile]: Unintentional indentation?

**Comment by daviditen** [comment]
Thanks @ben-albrecht!

**Pull request merged**